nautobot/docs/user-guides/custom-fields.md

```diff
@@ -0,0 +1,344 @@
+# Custom Fields
+
+Custom fields are a method of adding new fields to existing Nautobot models. For more general information on how custom fields work, refer to the [custom fields model documentation](../models/extras/customfield.md).
+
+## When to use Custom Fields
+
+Custom fields are commonly used for fields that need different values across individual objects. For example, a custom field on devices to reference an internal ticket number that inventories each device in Nautobot. If you need a solution for marking multiple objects with a common flag, [tags](../models/extras/tag.md) may be a better fit.
+
+## Create a Custom Field
+
+Navigate to the custom fields page by clicking on **Extensibility -> Custom Fields** in the Nautobot menu. Click on **Add** to create a new custom field.
+
+!!! note
+    Custom fields are initialized on associated objects when a content type is added to the custom field, including when the custom field is created. The initial value will be set to the `default` value of the custom field.
```

Comment:
From a developer perspective I understand this note but I don't think it's super clear to an end user. Perhaps:

1) Move this down to the `Default` or "Content Types" sub-sections below?
2) Rephrase to something like "When a custom field is created or associated to a new content type (model), all affected existing objects will be updated to include the custom field, receiving its `default` value. This update runs as a background task via Celery (insert link here), so it may take a few seconds or more before all objects reflect the new custom field, depending on the size of your database."